script.js

```diff
@@ -1,4 +1,273 @@
+  S.playerScores[S.pointerX].score += score
+  
+  message += `Player ${S.playerScores[S.pointerX].playerNum} rolls ${roll} and scores ${score}.<br>`
+  
+  S.pointerX += 1
+  if (S.pointerX === S.numPlayers) {
+    S.pointerX = 0
+    S.round += 1
+  }
+  
+  message += `<br>Player ${S.playerScores[S.pointerX].playerNum}, roll the dice.`
+
+  return message
+  
+  // fallthrough
+  return `🤦🏻‍♂️ from getGamePrompt()`
+
+}
+
+
+var getLeaderBoard = function() {
+  
+  var output = `🥇🥈🥉<br><br>`
+  
+  // upon each player having rolled dice, output leaderboard
+  if (S.knockoutMode) {
+    for (var i = 0; i < S.numPlayers; i += 1) {
+      if (S.players.includes(i)) {
+        output += `<small>Player ${i + 1}</small><br>`
+      } else {
+      output += `<del><small>Player ${i + 1}</small></del><br>`
+      }
+    }
+    return output
+  }
+
+  // sort players array by score
+  if (S.highScoreMode === true) S.playerScores.sort((a, b) => b.score - a.score)
+  else S.playerScores.sort((a, b) => a.score - b.score)
+
+  // output scores
+  for (var i = 0; i < S.playerScores.length; i += 1) {
+    if (S.playerScores[i].score === 0) continue
+    if (S.playerScores[i].score === S.playerScores[0].score)
+    output += `<small>Player ${S.playerScores[i].playerNum}: <b>${
+      S.playerScores[i].score
+    }</b></small><br>`
+    else
+    output += `<small>Player ${S.playerScores[i].playerNum}: ${
+      S.playerScores[i].score
+    }</small><br>`
+  }
+
+  S.playerScores.sort((a, b) => a.playerNum - b.playerNum) //resort players by playernum
+  return output
+}
+
+var getPlayerRolls = function() {
+  var output = `<b>Round ${S.round} 🎲</b><br><br>`
+
+  if (S.knockoutMode) {
+    for (var player of S.playerRolls) {
+      output += `<small>Player ${player[0] + 1}: ${player[2]}</small><br>`
+    }
+    return output
+  }
+
+  for (var i = 0; i < S.playerRolls.length; i += 1) {
+    output += `<small>Player ${i + 1}: ${S.playerRolls[i]}</small><br>`
+  }
+  return output
+}
+
+
+var getGameFooter = function() {
+  return ""
+  // return `<br><br><hr><h6>Players: ${S.numPlayers}<br>
+  // Dice: ${S.numDice}<br>
+  // ${S.highScoreMode ? "High" : "Low"}<br>
+  // ${S.knockoutMode ? "Knockout" : "Running"}<br>
+  // S.players ${S.players}</h6>`
+}
+
+var getTable = function() {
+  return `<table>
+  <tr>
+    <td valign = top style = "width:200px">${getLeaderBoard()}</td>
+    <td style = "width:350px">${getGamePrompt() + getGameFooter()}</td>
+    <td valign = top>${getPlayerRolls()}</td>
+  </tr>
+  </table>`
+}
+
+var main = function () {
+  return getTable()
+}
+
+var imgURL = "https://static.straitstimes.com.sg/s3fs-public/styles/article_pictrure_780x520_/public/articles/2021/09/19/rrsquidgame1909.jpg?itok=5RH3kEjV&timestamp=1632028214"
+document.body.style.backgroundImage = `url(${imgURL})`
+document.body.style.backgroundRepeat = "no-repeat"
+document.body.style.backgroundSize = "cover"
+document.getElementById("container").style.opacity = "0.85"
```

Comment:
these can be considered global variables / assignments which should go at the top of the script.js file